### PR TITLE
feat: add deposit gap check

### PR DESCRIPTION
**Pull request by amateima**
This PR adds deposit gap detection in the Across scraper.

**How it works:**
1. There's a new [deposit_gap_check](https://github.com/across-protocol/scraper-api/pull/368/files#diff-8359bea7a41a52f7fd73f2d59269c452b53ea0b924d477138c705a895afbdad4) table which stores the last deposit id that passes the gap check, meaning that they don't have gaps preceding them. This is useful to detect missing deposit events that need to be re-scraped.
2. There's a new [DepositGapService](https://github.com/across-protocol/scraper-api/pull/368/files#diff-1ed0239b8ef38c1d46359b1f174359ec4e31f67a1e4544ca9497427b02e7b1e1) which contains useful the logic for managing deposits gaps. 
The most important function is `checkDepositGaps()` that returns and object containing the following properties: 
- gapIntervals - a list of block gaps intervals `{ fromDepositId, toDepositId }`  
- lastDepositId - the max deposit id in the DB
- gapCheckPassDepositId - the max `depositId` that passes the gap check
3. The automation of deposit gap detection is done using [DepositsGapDetectionCron](https://github.com/across-protocol/scraper-api/pull/368/files#diff-c6facc94899d47679fd67ad1f69d763f3aec8bfebc11b5d962a80dfde838eabe) cron that runs every 10 minutes.

⚠️ For now this cron only checks for gaps. After running for some time in production, the next step is to add automatic re-scraping of the blocks containing missing deposits.

4. Each time a gap is detected a message is posted in the `zbot-across-scraper-production` Slack channel

Screenshot 2024-03-21 at 17 38 31

### CODE REVIEW

File: src/modules/scraper/service/DepositsGapDetectionCron.ts [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        ],
+      });
+      const bullets = [];
+      for (const gapInterval of gapIntervals) {
+        bullets.push({
+          type: "rich_text_section",
+          elements: [
+            {
+              type: "text",
+              text: `deposits from ${gapInterval.fromDepositId} to ${gapInterval.toDepositId}\n(previous: ${gapInterval.previousKnownDeposit?.depositId} ${gapInterval.previousKnownDeposit?.blockNumber}, next: ${gapInterval.nextKnownDeposit?.depositId} ${gapInterval.nextKnownDeposit?.blockNumber}`,
```
Comment (dohaki) [inline src/modules/scraper/service/DepositsGapDetectionCron.ts:130]: NIT: Maybe we can add more labels in the text body? It wasn't clear to me which number is the deposit id and which the block number by only looking at the slack message

**Review by dohaki — APPROVED** [review]
LGTM 👍 Left one small question and one nit

File: src/modules/scraper/model/DepositGapCheck.entity.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ */
+@Entity()
+export class DepositGapCheck {
+  @PrimaryColumn()
+  originChainId: number;
+
+  @PrimaryColumn()
+  depositId: number;
+
+  @Column()
```
Comment (james-a-morris) [inline src/modules/scraper/model/DepositGapCheck.entity.ts:14]: Are these implicitly non-nullable?

File: src/modules/scraper/service/DepositsGapDetectionCron.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+  private lock = false;
+
+  constructor(
+    private appConfig: AppConfig,
+    @InjectRepository(Deposit) private depositRepository: Repository<Deposit>,
+    private monitoringService: MonitoringService,
+    private depositGapService: DepositGapService,
+  ) {}
+
+  @EnhancedCron("0 5/10 * * * *")
```
Comment (james-a-morris) [inline src/modules/scraper/service/DepositsGapDetectionCron.ts:28]: OOC - why is this crontab run `At minute 0 past every 10th hour from 5 through 23.`?

**Review by james-a-morris — COMMENTED** [review]
Two questions

File: src/modules/scraper/service/DepositGapService.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  toDepositId: number;
+};
+
+type CheckDepositGapsResult = {
+  gapIntervals: DepositGapInterval[];
+  gapCheckPassDepositId?: number;
+  lastDepositId?: number;
+};
+
+const DEPOSITS_GAP_DETECTION_LIMIT = 50;
```
[the remark this answers is not in the document]
Comment (amateima) [reply src/modules/scraper/service/DepositGapService.ts:19]: Yeah

File: src/modules/scraper/service/DepositsGapDetectionCron.ts (hunk 1 above)
Comment (amateima) [reply src/modules/scraper/service/DepositsGapDetectionCron.ts]: done

File: src/modules/scraper/model/DepositGapCheck.entity.ts (hunk 2 above)
Comment (amateima) [reply src/modules/scraper/model/DepositGapCheck.entity.ts:14]: Yes

File: src/modules/scraper/service/DepositsGapDetectionCron.ts (hunk 3 above)
Comment (amateima) [reply src/modules/scraper/service/DepositsGapDetectionCron.ts:28]: it is running every ten minutes starting with minute 5

**Pull request merged**